### PR TITLE
Homework 04. Task 1 - 4

### PR DESCRIPTION
**Pull request by B0RlS**

### CODE REVIEW

File: HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_01.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  return [1] if numb == 1
+
+  fibo_array = [1, 1]
+  (2..numb - 1).each do |index|
+    fibo_array[index] = fibo_array[index - 2] + fibo_array[index - 1]
+  end
+  fibo_array
+end
+
+def check_validation(numb)
```
Comment (labs-ruby) [inline HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_01.rb]: def input_data_valid?(numb)

File: HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_01.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  return false if numb < 0 || numb > 1000
+  return false unless numb.class == Integer
+
+  true
```
Comment (labs-ruby) [inline HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_01.rb]: just numb.class == Integer

Comment (labs-ruby) [reply HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_01.rb]: or numb.kind_of? Integer

File: HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_04.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+def task_4_4(str)
+  str.downcase == str.downcase.reverse if chack_validation(str) == true
+end
+
+def chack_validation(input)
```
Comment (labs-ruby) [inline HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_04.rb]: chack - did you mean chuck_norris_strong_validation?

File: HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_04.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+def task_4_4(str)
+  str.downcase == str.downcase.reverse if chack_validation(str) == true
```
Comment (labs-ruby) [inline HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_04.rb]: if chack_validation(str) without == true
...
if chack_validation(str)
...
or better if  chuck_norris_strong_validation?

File: HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_04.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+end
+
+def chack_validation(input)
+  if input.class == String
+    true
+  else
+    puts 'Incorrect data entered'
+    false
+  end
+end
```
Comment (labs-ruby) [inline HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_04.rb]: input.kind_of?(String)

File: HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_04.rb (hunk 3 above)
Comment (B0RlS) [reply HW_04/Boris_Tsarikov/boris_tsarikov_hw_04_t_04.rb]: yeees)))

**Pull request merged**